DOC alphas_ documentation of LassoLarsIC

marenwestermann opened the pull request:
#### Reference Issues/PRs 
References #14312 


#### What does this implement/fix? Explain your changes.
Documents the attribute alphas_ in class LassoLarsIC

#### Any other comments?
I copied the code from the function `lars_path` in `_least_angle.py`. Since this documentation is the most up to date I also updated the documentation of the `alphas_` attributes in the classes `Lars` and `LassoLars`.

glemaitre reviewed (commented):
You need to remove `LassoLarsIC` from the list there: https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/tests/test_docstring_parameters.py#L236 to be sure that we are testing it.

marenwestermann committed:
correct shape descriptions of alphas_ attributes

marenwestermann committed:
remove LassoLarsIC from ignored list

marenwestermann commented:
@glemaitre Thank you for taking the time to review this PR. I have addressed your comments. I think I previously accidentally resolved a merge conflict the wrong way. I integrated the recent changes to the description of the shape of the alphas_ attributes in Lars and LassoLars into the documentation of the alphas_ attribute in LassoLarsIC.

glemaitre commented on sklearn/linear_model/_least_angle.py:
```diff
@@ -1765,6 +1766,13 @@ class LassoLarsIC(LassoLars):
     alpha_ : float
         the alpha parameter chosen by the information criterion
 
+    alphas_ : array-like of shape (n_alphas + 1,) or list of thereof of shape \
+            (n_targets,)
```
```suggestion
    alphas_ : array-like of shape (n_alphas + 1,) or list thereof
```

glemaitre commented on sklearn/linear_model/_least_angle.py:
```diff
@@ -1765,6 +1766,13 @@ class LassoLarsIC(LassoLars):
+        Maximum of covariances (in absolute value) at each iteration.
+        ``n_alphas`` is either ``max_iter``, ``n_features`` or the
+        number of nodes in the path with ``alpha >= alpha_min``, whichever
+        is smaller.
```
```suggestion
        is smaller. If a list, it will of length `n_targets`.
```

glemaitre reviewed (commented):
After the discussion in https://github.com/scikit-learn/scikit-learn/pull/18243, you only need to change a bit one of the docstring (the others will be changed within the previous PR for consistency).

marenwestermann committed:
minor docstring changes for consistency with #18243

glemaitre commented:
Thanks @marenwestermann all good
